File: src/main/java/clark/trackerapi/Data/Max.java
package clark.trackerapi.data;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Document
@Data
@AllArgsConstructor
public class Max {
    @Id
    private String id;
    @NonNull
    private LocalDate date;

    private Float squat1RM;
    private Float bench1RM;
    private Float deadlift1RM;
    private Float press1RM;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public Float getSquat1RM() {
        return squat1RM;
    }

    public void setSquat1RM(Float squat1RM) {
        this.squat1RM = squat1RM;
    }

    public Float getBench1RM() {
        return bench1RM;
    }

    public void setBench1RM(Float bench1RM) {
        this.bench1RM = bench1RM;
    }

    public Float getDeadlift1RM() {
        return deadlift1RM;
    }

    public void setDeadlift1RM(Float deadlift1RM) {
        this.deadlift1RM = deadlift1RM;
    }

    public Float getPress1RM() {
        return press1RM;
    }

    public void setPress1RM(Float press1RM) {
        this.press1RM = press1RM;
    }
}

File: frontend/src/data/max.ts
export interface IMax {
    id: string;
    date: string;
    squat1RM?: string;
    bench1RM?: string;
    deadlift1RM?: string;
    press1RM?: string;
}

export default class Max implements IMax {
    public id = "";
    public date = "";
    public squat1RM: string | undefined;
    public bench1RM: string | undefined;
    public deadlift1RM: string | undefined;
    public press1RM: string | undefined;

    public constructor(data: IMax) {
        this.id = data.id;
        this.date = data.date;
        this.squat1RM = data.squat1RM;
        this.bench1RM = data.bench1RM;
        this.deadlift1RM = data.deadlift1RM;
        this.press1RM = data.press1RM;
    }
}
